subxt/src/rpc/lightclient/mod.rs

```diff
@@ -0,0 +1,241 @@
+            database_content: None,
+            url: None,
+            bootnodes: None,
+            potential_relay_chains: None,
+        }
+    }
+}
+
+impl<'a> LightClientBuilder<'a> {
+    /// Create a new [`LightClientBuilder`].
+    pub fn new() -> LightClientBuilder<'a> {
+        LightClientBuilder::default()
+    }
+
+    /// The most important field of the configuration is the chain specification.
+    /// This is a JSON document containing all the information necessary for the client to
+    /// connect to said chain.
+    ///
+    /// The chain spec must be obtained from a trusted entity.
+    ///
+    /// It can be fetched from a trused node with the following command:
+    /// ```bash
+    /// curl -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "sync_state_genSyncSpec", "params":[true]}' http://localhost:9944/ | jq .result > res.spec
+    /// ```
+    ///
+    /// # Note
+    ///
+    /// For testing environments, please populate the "bootNodes" if the not already provided.
+    ///
+    /// ```json
+    ///   "bootNodes": [
+    ///       "/ip4/127.0.0.1/tcp/30333/p2p/12D3KooWEyoppNCUx8Yx66oV9fJnriXwCcXwDDUA2kj6vnc6iDEp"
+    ///    ],
+    /// ```
+    pub fn chain_spec(mut self, chain_spec: &'a str) -> Self {
+        self.chain_spec = Some(chain_spec);
+        self
+    }
+
+    /// Overwrite the bootnodes of the chain specification.
+    ///
+    /// Can be used to provide trusted entities to the chian spec, or for
+    /// testing environments.
+    pub fn bootnodes(mut self, bootnodes: impl Iterator<Item = &'a str>) -> Self {
+        self.bootnodes = Some(bootnodes.map(Into::into).collect());
+        self
+    }
+
+    /// Maximum number of JSON-RPC in the queue of requests waiting to be processed.
+    /// This parameter is necessary for situations where the JSON-RPC clients aren't
+    /// trusted. If you control all the requests that are sent out and don't want them
+    /// to fail, feel free to pass `u32::max_value()`.
+    ///
+    /// Default is 128.
+    pub fn max_pending_requests(mut self, max_pending_requests: NonZeroU32) -> Self {
+        self.max_pending_requests = max_pending_requests;
+        self
+    }
+
+    /// Maximum number of active subscriptions before new ones are automatically
+    /// rejected. Any JSON-RPC request that causes the server to generate notifications
+    /// counts as a subscription.
+    ///
+    /// Default is 1024.
+    pub fn max_subscriptions(mut self, max_subscriptions: u32) -> Self {
+        self.max_subscriptions = max_subscriptions;
+        self
+    }
+
+    /// After a chain has been added, it is possible to extract a "database" (in the form of a
+    /// simple string). This database can later be passed back the next time the same chain is
+    /// added again.
+    /// A database with an invalid format is simply ignored by the client.
+    pub fn database_content(mut self, database_content: &'a str) -> Self {
```

Comment:
Is this like a saved data file or something which can allow smoldot to save some work, I wonder?

There's no way to get back the database content, so I guess it may or may not be useful to expose this :)